lib/metasploit/framework/login_scanner/base.rb

```diff
@@ -57,8 +57,8 @@ module Base
                     inclusion: { in: [true, false] }
 
           validates :bruteforce_speed,
-                    presence: false,
                     numericality: {
+                      allow_nil: true,
```

Comment:
The numericality validator does everything you need for a number. take a look at the following example
```
validates :content
  format: { with: /\A[a-zA-Z]+\z/,
    message: "only allows letters" },
  length: {
    minimum: 300,
    maximum: 400,
    tokenizer: lambda { |str| str.split(/\s+/) },
    too_short: "must have at least %{count} words",
    too_long: "must have at most %{count} words"
  },
  unless: "content.nil?"
```
if content is nil you don't hit any of the validators, but if its not nil you will get the right error message for each of the validator.